docs/protocols/java_object_graph_spec.md

```diff
@@ -2,15 +2,26 @@
 
 ## Spec overview
 
-The data are serialized using little endian order overall. If bytes swap is costly, the byte order will be encoded as a
-flag in data.
+Fury Java Serialization is an automatic object serialization framework that supports reference and polymorphism. Fury
+will
+convert an object from/to fury java serialization binary format. Fury has two core concepts for java serialization:
+
+- **Fury Java Binary format**
+- Framework to convert object to/from Fury Java Binary format
 
-The overall format are:
+The serialization format is a dynamic binary format. The dynamics and reference/polymorphism support make Fury flexible,
+much more easy to use, but
+also introduce more complexities compared to static serialization frameworks. So the format will be more complex.
+
+Here is the overall format:
 
 ```
 | fury header | object ref meta | object class meta | object value data |
 ```
 
+The data are serialized using little endian order overall. If bytes swap is costly, the byte order will be encoded as a
```

Comment:
I think the statement should like this, should be more clear:

```diff
- the byte order will be encoded as a flag in data.
+ we can define the endian flag in Fury header
```

Reply:
How about this:
```
The data are serialized using little endian byte order overall. If bytes swap is costly for some object, 
Fury will write the byte order for that object into the data instead of converting it to little endian.
```